src/Bicep.Core/Registry/Oci/OciArtifactReference.cs

```diff
@@ -70,14 +70,18 @@ public class OciArtifactReference : ArtifactReference, IOciArtifactReference
         public override bool IsExternal => true;
 
         // unqualifiedReference is the reference without a scheme or alias, e.g. "example.azurecr.invalid/foo/bar:v3"
-        public static ResultWithDiagnostic<OciArtifactReference> TryParseModule(string? aliasName, string unqualifiedReference, RootConfiguration configuration, Uri parentModuleUri)
+        // The configuration and parentModuleUri are needed to resolve aliases and experimental features
+        public static ResultWithDiagnostic<OciArtifactReference> TryParseModuleAndAlias(string? aliasName, string unqualifiedReference, RootConfiguration configuration, Uri parentModuleUri)
```

Comment:
How is this function exercised in production code, all I can find are test references.

Reply:
Funny, I just added one in 61ffaca34f4809461276b4a8fa9d1331b70266cf.

Here though I was just trying to make the functions a little less confusing.  Separation of concerns for this class is currently very poor.  I'm improved it a little bit a while ago, but it could still use more, but I'd rather not deal with it now.